api/python/cellxgene_census/src/cellxgene_census/experimental/ml/pytorch.py

```diff
@@ -96,53 +99,83 @@ def __add__(self, other: "Stats") -> "Stats":
 
 @contextmanager
 def _open_experiment(
-    uri: str, aws_region: Optional[str] = None, soma_buffer_bytes: Optional[int] = None
+    uri: str,
+    aws_region: Optional[str] = None,
 ) -> soma.Experiment:
     """Internal method for opening a SOMA ``Experiment`` as a context manager."""
 
     context = _build_soma_tiledb_context(aws_region)
 
-    if soma_buffer_bytes is not None:
-        context = context.replace(
-            tiledb_config={
-                "py.init_buffer_bytes": soma_buffer_bytes,
-                "soma.init_buffer_bytes": soma_buffer_bytes,
-            }
-        )
-
     with soma.Experiment.open(uri, context=context) as exp:
         yield exp
 
 
 class _ObsAndXSOMAIterator(Iterator[_ObsAndXSOMABatch]):
-    obs_tables_iter: somacore.ReadIter[pa.Table]
-    """Iterates the SOMA batches (tables) of corresponding ``obs`` and ``X`` data. This is an internal class, 
+    """Iterates the SOMA batches of corresponding ``obs`` and ``X`` data. This is an internal class,
     not intended for public use."""
 
     X: soma.SparseNDArray
     """A handle to the full X data of the SOMA ``Experiment``"""
 
-    var_joinids: pa.Array
+    obs_joinids_chunks_iter: Iterator[npt.NDArray[np.int64]]
+
+    var_joinids: npt.NDArray[np.int64]
     """The ``var`` joinids to be retrieved from the SOMA ``Experiment``"""
 
-    def __init__(self, X: soma.SparseNDArray, obs_tables_iter: somacore.ReadIter[pa.Table], var_joinids: pa.Array):
-        self.obs_tables_iter = obs_tables_iter
+    def __init__(
+        self,
+        obs: soma.DataFrame,
+        X: soma.SparseNDArray,
+        obs_column_names: Sequence[str],
+        obs_joinids: npt.NDArray[np.int64],
+        var_joinids: npt.NDArray[np.int64],
+        soma_batch_read_size: int,
+        shuffle: bool = False,
+    ):
+        self.obs = obs
         self.X = X
+        self.obs_column_names = obs_column_names
+        self.obs_joinids_chunks_iter = self.chunk_obs_joinids(obs_joinids, soma_batch_read_size, shuffle)
```

Comment:
The SOMA reader for the obs DataFrame has been replaced with the list of `obs_joinids`. 
This was necessary to support global shuffling, since each soma chunk may comprise randomly-selected rows.  With the SOMA reader no longer determining the extent of each chunk, the chunking needs to be explicitly handled here now.